src/tools/miri/src/shims/intrinsics/mod.rs

```diff
@@ -131,6 +131,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 this.write_pointer(Pointer::new(ptr.provenance, masked_addr), dest)?;
             }
 
+            // The default implementation always returns `false`, but we want
+            // to return either `true` or `false` at random.
+            "is_comptile_time_known" => {
+                let rand = 0;
+                _ = getrandom::getrandom(&mut [rand]);
+
+                this.write_scalar(
+                    Scalar::from_bool(if (rand % 1) == 1 { true } else { false }),
+                    dest,
+                )?;
+            }
```

Comment:
I'm not sure if this is correct. I cannot test MIRI locally as there are *tons* of compile errors, so maybe somebody else can verify it?

Reply:
I thought of using `gen_random` instead but that seemed like a terrible idea, as this is a bool :sweat_smile:

Reply:
Doesn't this code always return `false` as `rand` is never modified?
```rust
if (rand % 1) == 1 { true } else { false }
```
is equivalent to
```rust
(rand % 1) == 1
```
which turns into
```rust
(0 % 1) == 1
```
which is `false`.

Reply:
So that's why it didn't need `mut`... I guess it's copied when the slice is created. Changing it to `&mut rand` and `[0]` would probably work fine

Reply:
I think you also need to `% 2` instead of `% 1` but otherwise that seems right.

Reply:
Yeah, I shouldn't code while half asleep. :D

Reply:
Please just use `this.machine.rng.get_mut().gen()`, which can return a `bool`.

Reply:
I have absolutely no idea how I missed that after searching `rng`...